node/subsystem/src/messages.rs

```diff
@@ -157,6 +210,16 @@ pub enum BitfieldDistributionMessage {
 	NetworkBridgeUpdate(NetworkBridgeEvent),
 }
 
+impl BitfieldDistributionMessage {
+	/// If the current variant contains the relay parent hash, return it.
+	pub fn relay_parent(&self) -> Option<Hash> {
+		match self {
+			Self::DistributeBitfield(hash, _) => Some(*hash),
+			Self::NetworkBridgeUpdate(_) => None,
+		}
+	}
+}
+
```

Comment:
I have a feeling that all these impls should be a trait

Reply:
I don't think so. Traits are for functionality you want to abstract over. I have a hard time imagining us writing code that is generic over `T: RelayParent` without some serious additional bounds. And then worse, once you've introduced a `trait RelayParent`, people have to import it every time they want to call `fn relay_parent`.